test/core/security/grpc_tls_crl_provider_test.cc

```diff
@@ -83,6 +101,110 @@ TEST(CrlProviderTest, StaticCrlProviderLookupBad) {
   EXPECT_EQ(crl, nullptr);
 }
 
+TEST(CrlProviderTest, DirectoryReloaderCrlLookupGood) {
+  auto result = experimental::CreateDirectoryReloaderProvider(
+      kCrlDirectory, std::chrono::seconds(1), nullptr);
+  ASSERT_TRUE(result.ok());
+  std::shared_ptr<CrlProvider> provider = std::move(*result);
+
+  CertificateInfoImpl cert = CertificateInfoImpl(kCrlIssuer);
+  auto crl = provider->GetCrl(cert);
+  ASSERT_NE(crl, nullptr);
+  ASSERT_EQ(crl->Issuer(), kCrlIssuer);
+
+  CertificateInfoImpl intermediate =
+      CertificateInfoImpl(kCrlIntermediateIssuer);
+  auto intermediate_crl = provider->GetCrl(intermediate);
+  ASSERT_NE(intermediate_crl, nullptr);
+  ASSERT_EQ(intermediate_crl->Issuer(), kCrlIntermediateIssuer);
+}
+
+TEST(CrlProviderTest, DirectoryReloaderCrlLookupBad) {
+  auto result = experimental::CreateDirectoryReloaderProvider(
+      kCrlDirectory, std::chrono::seconds(1), nullptr);
+  ASSERT_TRUE(result.ok());
+  std::shared_ptr<CrlProvider> provider = std::move(*result);
+
+  CertificateInfoImpl bad_cert = CertificateInfoImpl("BAD CERT");
+  auto crl = provider->GetCrl(bad_cert);
+  ASSERT_EQ(crl, nullptr);
+}
+
+std::string MakeTempDir() {
+  char templ[] = "/tmp/tmpdir.XXXXXX";
+  std::string dir_path = mkdtemp(templ);
+  return dir_path;
+}
+
+std::string TempDirNameFromPath(absl::string_view dir_path) {
+  std::vector<std::string> split = absl::StrSplit(dir_path, "/");
```

Comment:
This can use `absl::string_view` to avoid making unnecessary copies.

Reply:
Changed